.github/workflows/quick-test.yml

```diff
@@ -179,19 +179,23 @@ jobs:
   #       shell: cmd
   #       run: |
   #         C:\tools\cygwin\bin\bash -lc 'export PATH=/usr/local/bin:/usr/bin:/bin; cd /cygdrive/d/a/capnproto/capnproto; ./super-test.sh quick'
-  Linux-bazel:
+  Linux-bazel-clang:
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
-        compiler: [clang-10]
+        clang_version: [10, 14, 16]
     steps:
       - uses: actions/checkout@v3
       - uses: bazelbuild/setup-bazelisk@v2
       - name: install dependencies
         run: |
             export DEBIAN_FRONTEND=noninteractive
-            sudo apt-get install -y build-essential git ${{ matrix.compiler }}
+            sudo apt-get install -y build-essential git
+      - name: install clang
+        uses: egor-tensin/setup-clang@v1
```

Comment:
I'm not super excited about depending on random third-party repos, especially for something we only need temporarily (until a clang-16 package lands in Ubuntu, presumably).

Reply:
Actually I see this build is using Ubuntu 20.04, which is pretty old. Did you try 22.04? Maybe it has a clang-16 package already?

Reply:
OK I see... there is a clang-16 package but only for Ubuntu 23.04 (not LTS) so I guess indeed we'd have to wait another year.

I guess our actions don't have any secrets so there's no security concern here.

Given that we already cover clang 6 and 14 in the other builds, can we have the bazel build focus on 16? I worry we have too many builds...

Reply:
done. Bazel is clang-16 only.